src/libraries/System.Security.Cryptography.ProtectedData/src/System/Security/Cryptography/ProtectedData.cs

```diff
@@ -29,23 +173,174 @@ public static byte[] Unprotect(byte[] encryptedData, byte[]? optionalEntropy, Da
             if (encryptedData is null)
                 throw new ArgumentNullException(nameof(encryptedData));
 
-            return ProtectOrUnprotect(encryptedData, optionalEntropy, scope, protect: false);
+            byte[]? outputData;
+            bool result = TryProtectOrUnprotect(
+                encryptedData,
+                optionalEntropy,
+                scope,
+                protect: false,
+                allocateArray: true,
+                bytesWritten: out _,
+                outputData: out outputData
+            );
+
+            Debug.Assert(result);
+            Debug.Assert(outputData != null);
+            return outputData;
+        }
+
+        /// <summary>
+        /// Decrypts the data in a specified byte array and returns a byte array that contains the decrypted.
+        /// </summary>
+        /// <param name="encryptedData">A byte span that contains data to decrypt.</param>
+        /// <param name="scope">One of the enumeration values that specifies the scope of encryption.</param>
+        /// <param name="optionalEntropy">
+        /// An optional additional byte span used to increase the complexity of the encryption,
+        /// or empty for no additional complexity.
+        /// </param>
+        /// <returns>A byte array representing the encrypted data.</returns>
+        /// <exception cref="CryptographicException">The encryption failed.</exception>
+        /// <exception cref="NotSupportedException">The operating system does not support this method.</exception>
+        /// <exception cref="OutOfMemoryException">The system ran out of memory while encrypting the data.</exception>
```

Comment:
The existing Unprotect version is uninspiring (`Out of memory.`).  This one is OK, but has the wrong polarity.

```suggestion
        /// <exception cref="OutOfMemoryException">The system ran out of memory while decrypting the data.</exception>
```